src/meadowdb/readerwriter_shared.py

```diff
@@ -102,17 +103,30 @@ class TableSchema:
     deduplication_keys: List[str] = field(default_factory=list)
 
 
-@dataclass(frozen=True)
-class DataFileEntry:
+class TableLogEntry(abc.ABC):
```

Comment:
Can't add a comment there, but on line 82, the comment references DataFileEntry. Maybe worth changing the name of `data_list_filename`, also?